cli/command/telemetry_utils.go

```diff
@@ -0,0 +1,215 @@
+		"--version", "--login",
+		// added for build
+		"--builder", "--platform",
+		// added for testing things
+		"--availability",
+	}
+
+	managementCommands = []string{
+		"builder",
+		"checkpoint",
+		"container",
+		"context",
+		"image",
+		"manifest",
+		"network",
+		"plugin",
+		"system",
+		"trust",
+		"volume",
+	}
+
+	commands = []string{
+		"build",
+		"exec",
+		"images",
+		"info",
+		"login",
+		"logout",
+		"ps",
+		"pull",
+		"push",
+		"run",
+		"search",
+		"version",
+		"update",
+	}
+
+	orchestrationCommands = []string{
+		"config",
+		"node",
+		"secret",
+		"service",
+		"stack",
+		"swarm",
+	}
+)
+
+// GetCommand gets the invoked command as a string
+func GetCommand(args []string) string {
+	result := ""
+	onlyFlags := false
+	for _, arg := range args {
+		if arg == "--help" {
+			result = strings.TrimSpace(arg + " " + result)
+			continue
+		}
+		if arg == "--" {
+			break
+		}
+		if isCommandFlag(arg) || (!onlyFlags && isCommand(arg)) {
+			result = strings.TrimSpace(result + " " + arg)
+			if isCommand(arg) && !isManagementCommand(arg) && !isOrchestrationCommand(arg) {
+				onlyFlags = true
+			}
+		}
+	}
+	return result
+}
+
+// GetCommandNoFlags gets the invoked command as a string, without the flags
+func GetCommandNoFlags(args []string) string {
+	var res string
+	for _, arg := range strings.Split(GetCommand(args), " ") {
+		if strings.HasPrefix(arg, "-") {
+			continue
+		}
+		res = strings.TrimSpace(res + " " + arg)
+	}
+	return res
+}
+
+// stringContains checks if a string array contains a specific value
+func stringContains(array []string, needle string) bool {
+	for _, val := range array {
+		if val == needle {
+			return true
+		}
+	}
+	return false
+}
+
+func isOrchestrationCommand(word string) bool {
+	return stringContains(orchestrationCommands, word)
+}
+
+func isManagementCommand(word string) bool {
+	return stringContains(managementCommands, word)
+}
+
+func isCommand(word string) bool {
+	return stringContains(commands, word) || isManagementCommand(word) || isOrchestrationCommand(word)
+}
+
+func isCommandFlag(word string) bool {
+	return stringContains(commandFlags, word)
+}
+
+// otelErrorType returns an attribute for the error type based on the error category.
+// If nil, this function returns an invalid attribute.
+func otelErrorType(err error) string {
+	name := "generic"
+	if errors.Is(err, context.Canceled) {
+		name = "canceled"
+	}
+	return name
+}
+
+// BaseMetricAttributes returns an attribute.Set containing attributes to attach to metrics/traces
+func BaseMetricAttributes() attribute.Set {
+	attrList := []attribute.KeyValue{}
+
+	if command := GetCommand(os.Args); command != "" {
+		// TODO(krissetto): find proper way to get this info, this is just for testing
+		attrList = append(attrList, attribute.String("command.name", command))
+	}
+
+	return attribute.NewSet(attrList...)
+}
+
+func mapExitCode(exitCode int) string {
+	result := ""
+
+	switch exitCode {
```

Comment:
will we have more exit codes mapped here with different results? else we could just us an if statement like so

```go
if exitCode == 0 {
  return "success"
}

return "failure"
```

Reply:
Currently there is no need for multiple values, but we definitely could be adding more. I can change this for now